### PR TITLE
Fix reconnection issue in cli

### PR DESCRIPTION
**Pull request by haslinghuis**
On master it is not possible to go to `OSD` or `Sensor` tab from `CLI`.

This PR should solve this by using an interval to check for a valid connection and removing the duplicate call to the tab cleanup function by removing the call in the disconnect routine in `$('div.connect_controls a.connect')` and only use the call in the tab change request in `$('a', ui_tabs)`.

EDIT:
After lots of testing and input from @Asizon digging further the issue is sending or disconnecting when device_lost has occurred. Rebasing the serial code did solve the issue and some bugs were fixed.   

Tested on Matek F411 on Linux (Fedora 32) and Matek F722 on Windows (10)

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by Asizon** [comment]
@haslinghuis behaviour here is a litle weird, sometimes debug shows "CLI closed", "device rebooting", and sometimes it doesnt shows. Also when cli disconect reconect the tab is complety blank and doesnt goes to welcome tab.

**Comment by haslinghuis** [comment]
@Asizon Currently it only works with auto-connect enabled. Have it working now locally, needed some additional hacks in serial and msp, but still testing and debugging.

**Comment by mikeller** [comment]
Interestingly I cannot reproduce this problem in the first place (on linux) - when in the CLI tab, clicking on either the OSD or the Sensors tab gets me there after the flight controller reboot. Is there a different way to reproduce this?

**Comment by haslinghuis** [comment]
It's happening on a Matek F411 on Fedora 32 and Windows 10 both on master with latest firmware.

**Comment by haslinghuis** [comment]
This is really weird. I removed all previous code and rebased serial based code. On the same machine switching between two branches in different terminals this still solves the issue.

**Review by Asizon — APPROVED** [review]
Wow!All works as expected now.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/serial.js
Comment (mikeller) [inline src/js/serial.js:441]: What is the reason for this change?

File: src/js/main.js
Comment (mikeller) [inline src/js/main.js:265]: Good find!

File: src/js/serial.js
Comment (mikeller) [inline src/js/serial.js:238]: What is the reason that we aren't using `checkChromeRuntimeError()` here?

File: src/js/serial.js
Comment (haslinghuis) [reply src/js/serial.js:441]: New syntax - doing exactly the same.

File: src/js/serial.js
Comment (haslinghuis) [reply src/js/serial.js:238]: Maybe not needed anymore but I should not hurt. Needed the code to have less overhead to improve timing.

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
